metrics/qos/evm/metrics.go

```diff
@@ -49,13 +58,92 @@ var (
 	//   - End-to-end request success rates
 	//   - Error types by JSON-RPC method and chain
 	//   - HTTP status code distribution
+	//   - Service degradation when random endpoint fallback is used
 	requestsTotal = prometheus.NewCounterVec(
 		prometheus.CounterOpts{
 			Subsystem: pathProcess,
 			Name:      requestsTotalMetric,
 			Help:      "Total number of requests processed by EVM QoS instance(s)",
 		},
-		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code"},
+		[]string{"chain_id", "service_id", "request_origin", "request_method", "success", "error_type", "http_status_code", "random_endpoint_fallback"},
```

Comment:
Let's make this a different counter per the discussion:

<img width="1061" height="823" alt="Image" src="https://github.com/user-attachments/assets/e41c3439-0581-4e66-9713-81aaf15b0721" />